opts/gpus.go

```diff
@@ -0,0 +1,112 @@
+package opts
+
+import (
+	"encoding/csv"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/pkg/errors"
+)
+
+// GpuOpts is a Value type for parsing mounts
+type GpuOpts struct {
+	values []container.DeviceRequest
+}
+
+func parseCount(s string) (int, error) {
+	i := -1
+	var err error
+	if s != "all" {
+		i, err = strconv.Atoi(s)
+		if err != nil {
+			err = errors.Wrap(err, "count must be an integer")
+		}
+	}
+	return i, err
+}
+
+// Set a new mount value
+func (o *GpuOpts) Set(value string) error {
+	csvReader := csv.NewReader(strings.NewReader(value))
+	fields, err := csvReader.Read()
+	if err != nil {
+		return err
+	}
+
+	req := container.DeviceRequest{Options: make(map[string]string), Capabilities: [][]string{{"gpu"}}}
+
+	// Set writable as the default
+	for _, field := range fields {
+		parts := strings.SplitN(field, "=", 2)
+		key := strings.ToLower(parts[0])
+
+		if len(parts) == 1 {
+			req.Count, err = parseCount(key)
+			if err != nil {
+				return err
+			}
+			continue
+		}
+
+		if len(parts) != 2 {
```

Comment:
I don't think it can happen, as you use `strings.SplitN(field, "=", 2)`  🤔
It can just be len(parts) == 1 if there is no `=`, or 2 (and the second part remains unsplit), right?